### PR TITLE
docs: add supported technolgies page

### LINKED ISSUE #52: Add Supported technologies to docs
Need to provide documentation on which technologies are supported.

Some examples of similar docs in other projects:
https://www.elastic.co/guide/en/apm/agent/ruby/current/supported-technologies.html
https://www.elastic.co/guide/en/apm/agent/go/current/supported-tech.html
https://www.elastic.co/guide/en/apm/agent/java/current/supported-technologies-details.html

### PR DESCRIPTION
**Pull request by vigneshshanmugam**
+ fixes elastic/apm-agent-js-base#52

### CODE REVIEW

File: docs/supported-technologies.asciidoc [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+The Elastic APM RUM JavaScript agent will automatically instrument the following to give you detailed performance metrics of your web page
+
+* Page load metrics
+* Load time of Static Assets
+* API requests (XMLHttpRequest and Fetch)
+
+The agent uses the https://developer.mozilla.org/en-US/docs/Web/API/Navigation_timing_API[Navigation Timing API] and [Resource Timing API] available in the browsers to instrument the page load performance and static assets load times.
+
+The agent uses <<custom-transactions, Custom Transactions API>> to instrument all the outgoing XHR and Fetch API requests from the web page to the backend servers and also adds support for <<distributed-tracing-guide, Distributed Tracing>>.
```
Comment (jahtalab) [inline docs/supported-technologies.asciidoc]: I think we should say something like: `The agent automatically captures all the outgoing http requests, by instrumenting both XHR and Fetch APIs ....`

File: docs/supported-technologies.asciidoc [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+[float]
+[[frameworks]]
+=== Frameworks
+
+The agent is not framework agnostic and therefore will not instrument metrics related to rendering time or mounting time of components. If you are interested in measuring custom timings on frameworks like React, Angular, vue etc., please use the <<apm-start-span, Span API>> and create custom spans instead.
```
[the remark this answers is not in the document]
Comment (vigneshshanmugam) [reply docs/supported-technologies.asciidoc]: damm i screwed that up 😄 totally

File: docs/supported-technologies.asciidoc [hunk 3]
```diff
@@ -0,0 +1,26 @@
+[[supported-technologies]]
+== Supported Technologies
+
+The Elastic APM RUM JavaScript agent will automatically instrument the following to give you detailed performance metrics of your web page
```
Comment (bmorelli25) [inline docs/supported-technologies.asciidoc]:
```suggestion
The Elastic APM RUM JavaScript agent gives you detailed performance metrics of your web page by automatically instrumenting the following:
```

File: docs/supported-technologies.asciidoc (hunk 1 above)
Comment (bmorelli25) [reply docs/supported-technologies.asciidoc]: Should probably also break distributed tracing into its own sentence to avoid a run on sentence

**Pull request merged**